### ISSUE TITLE
Allow --includeIPRanges to be passed to Istio

### ISSUE THREAD

**Issue by olhtbr** [datawire/forge#121]
The `istio kube-inject` command takes a couple of parameters, one that I think is very useful is the `--includeIPRanges` argument. It tells Istio to redirect outbound traffic to Envoy only for the given IP ranges. It would be great, if this parameter could be passed to Forge in the `service.yaml` file.

**Comment by rhs** [comment]
Good idea. I will add something for this next release.

**Comment by olhtbr** [comment]
I'd be happy to start working on it. I'm thinking of extending `istio` to a map in service.yaml, like this

```yaml
istio:
  enable: true
  includeIPRanges:
    - CIDR1
    - CIDR2
    ...
```

**Comment by rhs** [comment]
Go for it!

FYI, I recently rewrote dev instructions after being "inspired" by some unfortunate travis changes to create a less fragile dev/test/release environment. You can check them out here:

https://github.com/datawire/forge/blob/master/DEVELOPING.md

Any feedback on how well they work for you would be greatly appreciated!

**Comment by olhtbr** [comment]
After a little holiday delay, I finally opened #128, but I'm a little puzzled with the travis failure. Is there something I should fix?

PS. I'm a complete python beginner, so please bear with me :)

**Comment by olhtbr** [comment]
Oh and for the feedback: it worked really well, except istioctl. If I work in the developer shell (or bridge), then istioctl kube-inject does not work, it simply writes nothing to stdout. Maybe it's something with the container's shell..

As a result I could run my tests without the container and they passed, but when I started the full test suite with `dev.sh test`, the istio tests failed.

**Comment by rhs** [comment]
Thanks for the feedback!

I'm pretty sure the travis issue you are seeing is because you aren't in the datawire organization and so the kubernaut, aws, and gcr credentials aren't available on your PR for security reasons.

I'm currently just finishing up vacation myself. My plan (assuming my house has survived the storm in Boston) is to do a forge release next week that includes this and hopefully some other fixes as well. I will also look at the kube-inject issue and see if I can come up with a better way of dealing with testing credentials.

**Comment by rhs** [comment]
FYI, the PR is merged onto master and I should be doing a release in the next day or so...

BTW, I'd love to learn more about your usage of forge... we'll probably be doing some more extensive istio integration ourselves soon... also, any quote you'd be willing to share a quote for the forge.sh website would be greatly appreciated.

**Comment by olhtbr** [comment]
Great, thanks! :)

As for the usage, we started to use forge just recently during development. I also plan to introduce it for automated deployments from CI. IMHO the dependency handling is nice during development, but I don't think it will be used much during CI deployments, as I would like to deploy all services independently.

I started with [kubernetes-deploy](https://github.com/Shopify/kubernetes-deploy), which is also great, but felt a bit opinionated. Although it can wait for pods to actually start, which is very nice. Maybe something that could be supported by forge too in the future?

**Comment by rhs** [comment]
FYI, This has been released in 0.3.24

Regarding your feedback, thanks for taking the time. It sounds pretty consistent with our own experience. We are exploring how to better deal with collections of services.

I'm totally with you on the pod startup/feedback front, I've seen kubernetes-deploy before and definitely want to somehow cover that functionality in forge. I created a separate issue for this #135.
